### PR TITLE
Split lease task queue API into 2

### PR DESCRIPTION
**Pull request by wxing1292**
**What changed?**
* Split matching persistence LeaseTaskQueue API into
  * CreateTaskQueue
  * GetTaskQueue
* Add new task queue UT
* Add new task queue task UT
* Deprecate legacy matching task tests
* Modify logic accordingly

**Why?**
Code cleanup & test cleanup

**How did you test it?**
New tests

**Potential risks**
N/A

**Is hotfix candidate?**
No

### CODE REVIEW

**Comment by wxing1292** [comment]
review #2262 first

File: common/persistence/persistence-tests/persistenceTestBase.go [hunk 1]
```diff
@@ -67,6 +67,10 @@ import (
 	"go.temporal.io/server/service/history/tasks"
 )
 
+// TimePrecision is needed to account for database timestamp precision.
+// Cassandra only provides milliseconds timestamp precision, so we need to use tolerance when doing comparison
+const TimePrecision = 2 * time.Millisecond
```
Comment (yiminc) [inline common/persistence/persistence-tests/persistenceTestBase.go:72]: this is not used anywhere in this PR?

**Review by yiminc — COMMENTED** [review]
This PR include changes from 2263, should I hold on reviewing this until 2263 is landed?

File: common/persistence/persistence-tests/persistenceTestBase.go (hunk 1 above)
Comment (wxing1292) [reply common/persistence/persistence-tests/persistenceTestBase.go:72]: it is used by existing persistence tests, simply moving the definition from file to be deleted to here

File: service/matching/db.go [hunk 2]
```diff
@@ -88,34 +96,104 @@ func (db *taskQueueDB) RangeID() int64 {
...
+		if err := db.renewTaskQueue(db.rangeID + 1); err != nil {
+			return taskQueueState{}, err
+		}
 	}
-	db.ackLevel = resp.TaskQueueInfo.Data.AckLevel
-	db.rangeID = resp.TaskQueueInfo.RangeID
 	return taskQueueState{rangeID: db.rangeID, ackLevel: db.ackLevel}, nil
 }
 
+func (db *taskQueueDB) takeTaskQueue() error {
```
Comment (yiminc) [inline service/matching/db.go:112]: add Locked suffix, and maybe take -> takeover

File: service/matching/db.go [hunk 3]
```diff
@@ -88,34 +96,104 @@ func (db *taskQueueDB) RangeID() int64 {
...
+		}
+		db.rangeID = initialRangeID
+		return nil
+
+	default:
+		return err
+	}
+}
+
+func (db *taskQueueDB) renewTaskQueue(
```
Comment (yiminc) [inline service/matching/db.go:158]: add Locked suffix.

File: service/matching/matchingEngine_test.go [hunk 4]
```diff
@@ -391,7 +391,10 @@ func (s *matchingEngineSuite) PollForTasksEmptyResultTest(callContext context.Co
 	tl := "makeToast"
 	identity := "selfDrivingToaster"
 
-	taskQueue := &taskqueuepb.TaskQueue{Name: tl}
+	taskQueue := &taskqueuepb.TaskQueue{
+		Name: tl,
+		Kind: enumspb.TASK_QUEUE_KIND_NORMAL,
```
Comment (yiminc) [inline service/matching/matchingEngine_test.go:396]: the Unspecified should be treated as Normal.

**Review by yiminc — APPROVED** [review]

**Pull request merged**